api/v2alpha1/pipeline_spec.proto

```diff
@@ -74,7 +75,139 @@ message PipelineSpec {
     Value default_value = 2;
   }
   // The runtime parameters of the pipeline. Keyed by parameter name.
-  map<string, RuntimeParameter> runtime_parameters = 6;
+  // Deprecated, instead of defining the runtime_parameters, user can define
+  // [ComponentSpec.input_definitions][] field of the [PipelineSpec.root][] to
+  // define the pipeline parameters.
+  map<string, RuntimeParameter> runtime_parameters = 6 [deprecated = true];
+
+  // The map of name to definition of all components used in this pipeline.
+  map<string, ComponentSpec> components = 8;
+
+  // The definition of the main pipeline.  Execution of the pipeline is
+  // completed upon the completion of this component.
+  ComponentSpec root = 9;

```

Comment:
Just realized this can potentially look confusing once we introduce `pipeline_root` (or equivalent) concept for URI placeholder support. Perhaps some renaming can help here.

/cc @chensun 
/cc @neuromage 
I think we can consider renaming `pipeline_root` to something like `output_directory`. WDYT?

Reply:
> I think we can consider renaming pipeline_root to something like output_directory. WDYT?

SGTM.